adam-core/src/main/scala/org/bdgenomics/adam/util/FileExtensions.scala

```diff
@@ -63,7 +63,11 @@ private[adam] object FileExtensions {
    */
   def isFastqExt(pathName: String): Boolean = {
     pathName.endsWith(".fq") ||
-      pathName.endsWith(".fastq")
+      pathName.endsWith(".fastq") ||
+      pathName.endsWith(".fastq.gz") ||
```

Comment:
Yeah, good catch RE docs update and `ADAMContext.trimExtensionIfCompressed`. Due to the vagaries of how the BGZF codecs get set up, I'll need to modify `ADAMContext.trimExtensionIfCompressed` but it is a very small modification.

I'll also make a more extensive documentation pass.